test/gtest/uct/test_uct_ep.cc

```diff
@@ -85,4 +162,73 @@ UCS_TEST_SKIP_COND_P(test_uct_ep, disconnect_after_send,
     }
 }
 
+UCS_TEST_SKIP_COND_P(test_uct_ep, destroy_entity_after_send,
+                     !check_caps(UCT_IFACE_FLAG_AM_ZCOPY))
+{
+    const unsigned max_iter = 300 / ucs::test_time_multiplier();
+    char *buffer_mem        = new char[sizeof(mapped_buffer)];
+
+    for (unsigned i = 0; i < max_iter; ++i) {
+        connect(false);
+
+        mapped_buffer *buffer = new (buffer_mem)
+                mapped_buffer(256 * UCS_KBYTE, 0, *m_sender);
+        UCS_TEST_GET_BUFFER_IOV(iov, iovcnt, buffer->ptr(),
+                                ucs_min(buffer->length(),
+                                        m_sender->iface_attr().cap.am.max_zcopy),
+                                buffer->memh(),
+                                m_sender->iface_attr().cap.am.max_iov);
+
+        ucs_status_t status;
+
+        test_ep_comp_t comp(this, m_sender->ep(0), buffer, completion_cb);
+
+        for (unsigned count = 0; count < max_iter;) {
+            status = uct_ep_am_zcopy(m_sender->ep(0), 1, NULL, 0, iov, iovcnt,
+                                     0, &comp.comp);
+            if (status == UCS_INPROGRESS) {
+                ++comp.comp.count;
+            } else if (status != UCS_OK) {
+                EXPECT_EQ(UCS_ERR_NO_RESOURCE, status);
+                if (count == 0) {
+                    progress();
+                } else {
+                    status = uct_ep_flush(m_sender->ep(0), UCT_FLUSH_FLAG_LOCAL,
+                                          &comp.comp);
+                    if (status == UCS_INPROGRESS) {
+                        ++comp.comp.count;
+                    } else {
+                        EXPECT_EQ(UCS_ERR_NO_RESOURCE, status);
+                    }
+                    break;
+                }
+            }
+            ++count;
+        }
+
+        if (comp.comp.count == 0) {
+            /* Neither AM Zcopy nor flush operations were scheduled, release
+             * buffer and retry */
+            buffer->~mapped_buffer();
+            continue;
+        }
+
+        {
+            scoped_log_handler slh(detect_uncomp_op_logger);
+            disconnect(false);
+            /* All outstanding operations must be completed in EP destroy */
+            EXPECT_EQ(0, comp.comp.count);
+
+            m_entities.remove(m_sender);
+        }
+
+        /* Create new sender for new iteration or it will be destroyed in
+         * cleanup() */
+        m_sender = uct_test::create_entity(0);
+        m_entities.push_back(m_sender);
```

Comment:
create sender in the beginning of the loop

Reply:
IMO, better remove it from init for this test class

Reply:
done